### ISSUE TITLE
SimClr -CAMELYON16

### ISSUE THREAD

**Issue by Amcky** [binli123/dsmil-wsi#10]
Could you please post the weight of SIMCLR training on CAMELYON16 dataset

**Comment by binli123** [comment]
https://drive.google.com/drive/folders/1jd5qbpZ0fdqJdH3FYQaTimxYHkzNa_bW?usp=sharing
ResNet18 with instance normalization for the 20x patches (at level 1, level 0 is 40x).
I will also upload precomputed features for Camelyon16 soon.

**Comment by windstormer** [comment]
With the provided pretrain weight of SIMCLR, I had tried to compute the embeddings and apply to train_tcga.py (with dataset adaption).
However, the AUC results is about 0.66 which is far from the one presented in the paper.
Is there any setting modification for MIL parts for CAMELYON16 from TCGA dataset?
![image](https://user-images.githubusercontent.com/14368570/126664962-9d2f10a2-4db8-44ad-a8f8-9bdb2c0f4518.png)

**Comment by binli123** [comment]
> With the provided pretrain weight of SIMCLR, I had tried to compute the embeddings and apply to train_tcga.py (with dataset adaption).
> However, the AUC results is about 0.66 which is far from the one presented in the paper.
> Is there any setting modification for MIL parts for CAMELYON16 from TCGA dataset?
> ![image](https://user-images.githubusercontent.com/14368570/126664962-9d2f10a2-4db8-44ad-a8f8-9bdb2c0f4518.png)

Do you use the latest `train_tcga.py`? Your output doesn't seem right to me. You will need to set the --num_classes=1. I have updated the readme file.

**Comment by windstormer** [comment]
Currently, I found that your code will assign the label 1 for "normal" and 0 "tumor" in camelyon16 dataset, which is because the sorting algorithm did not follow alphabetic order.
You could replace "sorted(num_classes)" in line 231 of compute_feats.py to "num_classes.sort()".
Short suggestion.

**Comment by binli123** [comment]
> Currently, I found that your code will assign the label 1 for "normal" and 0 "tumor" in camelyon16 dataset, which is because the sorting algorithm did not follow alphabetic order.
> You could replace "sorted(num_classes)" in line 231 of compute_feats.py to "num_classes.sort()".
> Short suggestion.

https://discuss.codecademy.com/t/what-is-the-difference-between-sort-and-sorted/349679

**Comment by linzhenyuyuchen** [comment]
> With the provided pretrain weight of SIMCLR, I had tried to compute the embeddings and apply to train_tcga.py (with dataset adaption).
> However, the AUC results is about 0.66 which is far from the one presented in the paper.
> Is there any setting modification for MIL parts for CAMELYON16 from TCGA dataset?
> ![image](https://user-images.githubusercontent.com/14368570/126664962-9d2f10a2-4db8-44ad-a8f8-9bdb2c0f4518.png)

I also found that the AUC results is about 0.66.
Is there any solution ?
Thank you !

**Comment by windstormer** [comment]
> > With the provided pretrain weight of SIMCLR, I had tried to compute the embeddings and apply to train_tcga.py (with dataset adaption).
> > However, the AUC results is about 0.66 which is far from the one presented in the paper.
> > Is there any setting modification for MIL parts for CAMELYON16 from TCGA dataset?
> > ![image](https://user-images.githubusercontent.com/14368570/126664962-9d2f10a2-4db8-44ad-a8f8-9bdb2c0f4518.png)
> 
> I also found that the AUC results is about 0.66.
> Is there any solution ?
> Thank you !

As the author @binli123 had mentioned, you'll have to check if the negative bags had the label as 0 and the positive ones as 1.
By solving this problem, I can receive a over 0.9 AUC with provided weights.

**Comment by binli123** [comment]
Trained models are updated to this link now: https://drive.google.com/drive/folders/14pSKk2rnPJiJsGK2CQJXctP7fhRJZiyn?usp=sharing.
These models are trained with different settings and a shorter training time & smaller batch size will lead to representations that require longer converge time for the downstream MIL  networks.
